libs/sdk-ui-filters/src/AttributeFilter/utils/AttributeFilterUtils.ts

```diff
@@ -52,37 +51,36 @@ export const getNoneTitleIntl = (intl: IntlShape): string => {
 
 export const getItemsTitles = (
     selectedFilterOptions: IAttributeElement[],
-    validOptions: IAttributeElement[],
+    elementTitles: Map<string, IAttributeElement>,
 ): string => {
-    const validOptionsByUri = keyBy(validOptions, "uri");
-    const validOptionsByTitle = keyBy(validOptions, "title");
     return selectedFilterOptions
-        .map(
-            (selectedOption) =>
-                selectedOption.title ??
-                validOptionsByUri[selectedOption.uri]?.title ??
-                validOptionsByTitle[selectedOption.title]?.title,
+        .map((selectedOption) =>
+            selectedOption?.uri ? elementTitles.get(selectedOption.uri)?.title : undefined,
         )
         .join(", ");
 };
 
+export const updateSelectedOptionsWithDataByMap = (
+    selection: Array<Partial<IAttributeElement>>,
+    validElements: Map<string, IAttributeElement>,
+): Array<IAttributeElement> => {
+    return selection.map((selectedItem) => {
+        return validElements.get(selectedItem.uri);
+    });
+};
+
 export const updateSelectedOptionsWithData = (
```

Comment:
Are changes in this function needed anymore since the Map is used? I wonder if it will not break the other filters...

Reply:
It won't break any other filter as the original function didn't change and the new one was introduced to be used on the AttributeFilterButton